pkg/json/json.go

```diff
@@ -275,11 +275,15 @@ var base64JSONStr = func(json, arg string) string {
 
 	switch arg {
 	case "encode":
-		encoded := base64.URLEncoding.EncodeToString([]byte(str))
+		encoded := base64.StdEncoding.EncodeToString([]byte(str))
 		return fmt.Sprintf("\"%s\"", encoded)
 	case "decode":
-		decoded, _ := base64.URLEncoding.DecodeString(str)
-		return fmt.Sprintf("\"%s\"", decoded)
+		encoding := base64.StdEncoding
+		if !strings.HasSuffix(str, "=") {
```

Comment:
From the docs,

> StdEncoding is the standard base64 encoding, as defined in RFC 4648.

> RawStdEncoding is the standard raw, unpadded base64 encoding, as defined in RFC 4648 section 3.2.
> This is the same as StdEncoding but omits padding characters.

This means that `StdEncoding` only works if the base64-encoded input string is a multiple of 4 bytes; i.e., shorter strings must be padded. If the string length is not a multiple of 4 – i.e. it is not padded –, `RawStdEncoding` must be used instead. However, padded strings cannot be decoded using `RawStdEncoding`.

Using your own example to more or less demonstrate with the `base64` command (standard algorithm, i.e. requires padding):

```sh
❯ echo -n "hella" | base64
aGVsbGE=

❯ echo -n "aGVsbGE=" | base64 -d
hella%

❯ echo -n "aGVsbGE" | base64 -d
❯
```

Reply:
Yeah, my base64 binary is explicit showing the error:

```
❯ base64 --version
base64 (GNU coreutils) 8.32

❯ echo -n "aGVsbGE" | base64 -d
hellabase64: invalid input
```
I have refreshed my knowledge on base64 encoding. Two inputs:
a) the decoding error should be handled (feel free to do that in another PR) when the encoding is not valid
b) the current implementation searching for `=` padding char (btw which exists as constant in the golang package `StdPadding`) should work and I cannot find issue from that. However, I would like to consider two other implementation which can be used:

```go
rawByte, err := base64.StdEncoding.DecodeString(s)
if err != nil {
    rawByte, err = base64.RawStdEncoding.DecodeString(s)
}
```

or 

```
if len(s) % 4 != 0 {
    // not padded. so it cannot be base64.StdEncoding 
    rawByte, err = base64.RawStdEncoding.DecodeString(s)
} else {
    rawByte, err = base64.StdEncoding.DecodeString(s)
}
```

Reply:
Anyway, I would not expect not padded (when padding is necessary) base64 encoded inputs. So my first try would be use only ` rawByte, err = base64.StdEncoding.DecodeString(s)` and regard inputs not being multiple of 4 octets as errors :shrug:

Reply:
> the decoding error should be handled

Well, only if it's to log the error, but the modifier function itself has to be resilient. Even for logging we have to be careful as this is straight from user input.

Reply:
I'm OK with this implementation too, I guess:

```
if len(s) % 4 != 0 {
    // not padded. so it cannot be base64.StdEncoding 
    rawByte, err = base64.RawStdEncoding.DecodeString(s)
} else {
    rawByte, err = base64.StdEncoding.DecodeString(s)
}
```

Although the equal sign (`=`) is by specification. So I really don't see an issue with the current implementation either.

Reply:
> I would not expect not padded (when padding is necessary) base64 encoded inputs.

I'm afraid we have to expect unpadded values as well because of HTTP Basic Auth ([RFC 7617](https://www.rfc-editor.org/rfc/rfc7617)).

Reply:
@eguzki , WDYT about this option?

```go
var base64JSONStr = func(json, arg string) string {
	str := gjson.Parse(json).String()

	switch arg {
	case "encode":
		encoded := base64.StdEncoding.EncodeToString([]byte(str))
		return wrap(encoded)
	case "decode":
		if len(str) % 4 == 0 {
			if decoded, err := base64.StdEncoding.DecodeString(str); err == nil {
				return escapeAndWrap(string(decoded))
			}
		}
		// try raw standard encoding if unpadded or error
		decoded, _ := base64.RawStdEncoding.DecodeString(str)
		return escapeAndWrap(string(decoded))
	default:
		return json
	}
}

func wrap(s string) string {
	return fmt.Sprintf("\"%s\"", s)
}

func escapeAndWrap(s string) string {
	return wrap(strings.ReplaceAll(s, `"`, `\"`))
}
```